waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -264,12 +265,35 @@
                      :revision-timestamp {:pod revision-timestamp :rs rs-revision-timestamp}})
           true))))
 
+;; forward declaration of the hard-delete-service-instance function
```

Comment:
Okay, yeah, I see that now. Thanks for clarifying. I'm fine with just leaving the forward declaration as long as the comment is changed to note that it's only there to avoid reordering a bunch of functions (which also makes the git history messy for operations like git-blame).